java/src/main/java/io/cucumber/java/InvalidMethodSignatureException.java

```diff
@@ -54,34 +58,35 @@ public InvalidMethodSignatureException build() {
         }
 
         private String describeNote() {
-            return String.join("\n", notes);
+            return String.join("\n", this.notes);
```

Comment:
With the exception of constructors, inner classes and setters the `this` key word isn't usually needed.

Reply:
yep, like my other comments, due to previous projects it's easier to do this as having had bugs introduced by senior/experience developers who refactored code and created local variables with the same name as member variables so code then breaks, won't change existing code without a need to have this.

Reply:
reverted back